src/ad-sizes.test.ts

```diff
@@ -19,3 +19,20 @@ describe('ad sizes', () => {
 		},
 	);
 });
+
+describe('getTuple', () => {
+	it.each([
+		['mpu' as const, [300, 250]],
+		['fluid' as const, [0, 0]],
+		['googleCard' as const, [300, 274]],
+		['outstreamGoogleDesktop' as const, [550, 310]],
+		['video' as const, [620, 1]],
+		['300x600' as const, [300, 600]],
+	])(
```

Comment:
Not sure the `as const` assertion is doing anything for you here. You need to have a mutable object like an array of object to get more type safety, like so:

```ts
const sizes = [
		['mpu', [300, 250]],
		['fluid', [0, 0]],
		['googleCard', [300, 274]],
		['outstreamGoogleDesktop', [550, 310]],
		['video', [620, 1]],
		['300x600', [300, 600]],
	] as const;

it.each(sizes)(/* … */);
```

But personally I’d go for something like, so you get type safety as you build your object: 
```ts
const sizes: Array<[SizeKeys, AdSizeTupe]> = [
		['mpu', [300, 250]],
		['fluid', [0, 0]],
		/* etc… */
]
```

Reply:
Ah thank you, that's much better!